src/main/scala/Column.scala

```diff
@@ -1034,13 +1096,88 @@ case class Column private (
     }
   }
 
+  def validateFormat(value: String, rowNumber: Long): Boolean = {
+    format match {
+      case Some(f) => {
+        val formatValidator = datatypeFormatValidation(baseDataType)
+        if (formatValidator(value)) true
+        else {
+          errors = errors :+ ErrorWithCsvContext(
+            "format",
+            "schema",
+            rowNumber.toString,
+            columnOrdinal.toString,
+            value,
+            s"Value in csv does not match the format specified in metadata, Value: ${value} Format: ${f.pattern.get}"
+          )
+          false
+        }
+      }
+      case None => true
+    }
+  }
+
+  def validateLength(value: String, rowNumber: Long): Boolean = {
+    var valid = true
+    if (
+      datatype.path("length").isMissingNode && datatype
```

Comment:
Don't you want to use the `minLength` value (and others) that you defined earlier in the `Column` constructor?

Reply:
Done